### ISSUE TITLE
addCleanup is documented as being called after tearDown, but is actually called before

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3131]
|<img alt="arkanes's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| arkanes reported|
|-|-|
|Trac ID|trac#3131|
|Type|defect|
|Created|2008-03-27 19:37:52Z|
|Branch|https://github.com/twisted/twisted/tree/addCleanup-docstring-3131|

There's a specific test case for this behavior (test_tests.TestAddCleanup.test_addCleanupCalledInReverseOrder), so the docstring is wrong.

<details><summary>Searchable metadata</summary>

```
trac-id__3131 3131
type__defect defect
reporter__arkanes arkanes
priority__highest highest
milestone__ 
branch__branches_addCleanup_docstring_3131 branches/addCleanup-docstring-3131
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1206646672000000 1206646672000000
changetime__1206661863000000 1206661863000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [23108]) Branching to 'addCleanup-docstring-3131'

**Comment by twisted-trac** [comment]
|<img alt="jml-insecure's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|jml-insecure commented|
|-|-|

Made the correction. Also made a couple of other stylistic changes to the docstring.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

Time to make & publish the change: 3 minutes.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

No functional changes; go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [23110]) Fix the docstring for `addCleanup`.

 * Author: jml
 * Reviewer: glyph
 * Fixes [#3131](https://github.com/twisted/twisted/issues/3131)

The docstring for `addCleanup` incorrectly said that the cleanup functions
were called _after_ `tearDown`. The docstring now says the truth: that they
are run _before_ tearDown.

That is all.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
